packages/engine/Source/Scene/Scene.js

```diff
@@ -3834,8 +3831,22 @@ Scene.prototype.initializeFrame = function () {
   this._tweens.update();
 
   if (this._globeHeightDirty) {
+    this._removeUpdateHeightCallback =
+      this._removeUpdateHeightCallback && this._removeUpdateHeightCallback();
```

Comment:
This line seems confusing from a types perspective. Especially when paired with the line below where `updateHeight` sets it to a function.
Also I'm a little confused why we're setting the value here and then setting it again below? Is it used in the `getGlobeHeight` function? or is this trying to call it if it exists and set it below regardless?

Reply:
What is is doing is that, if this._removeUpdateHeightCallback is a defined function, execute the function (therefor removing the callback) and reassign the value to false.

It's removing the old listener (if it exists) before creating a new listener. This is a code pattern which is used commonly in the project.

Reply:
So it's equivalent to this?

```js
if (defined(this._removeUpdateHeightCallback)) {
  this._removeUpdateHeightCallback();
  this._removeUpdateHeightCallback = false;
} else {
  this._removeUpdateHeightCallback = false;
}

// or more concise

if (defined(this._removeUpdateHeightCallback)) {
  this._removeUpdateHeightCallback();
}
this._removeUpdateHeightCallback = false;
```

1. We normally avoid trying to do truthy/falsey checks, why not use `defined()` in the first `&&`?
2. Why assign to `false` as a boolean when it's also a function or `undefined`? Even if it's a common pattern this seems like asking for issues? Is `false` ever actually checked for? (I didn't see it anywhere)
    - Would it be better to do `delete this._removeUpdateHeightCallback`?